plugins/zfs/zpool_capacity

```diff
@@ -0,0 +1,259 @@
+#!/bin/bash
+
+: << =cut
+
+=head1 NAME
+
+  zpool_capacity - Munin plugin to monitor ZFS capacity
+
+  These functions are implemented:
+    capacity  : to monitor zpool capacity %
+    allocated : to monitor zpool allocated bytes
+    dedup     : to monitor zpool dedup and compress ratio
+
+  Tested with Solaris 10 and 11, OpenIndiana Hipster, FreeBSD 11, CentOS 7
+
+=head1 CONFIGURATION
+
+  Make symlink:
+    cd /path/to/munin/etc/plugins
+    ln -s /path/to/munin/lib/plugins/zpool_capacity .
+
+  For FreeBSD, it should be necessary to change shebang /bin/bash -> /usr/local/bin/bash
+
+  For Linux, root privilege is necessary to run zpool command.
+    [zpool_capacity]
+      user  root
+
+=head1 ENVIRONMENT VARIABLES
+ 
+  critical : default 90
+  warning  : default 80
+
+=head1 AUTHOR
+
+  K.Cima https://github.com/shakemid
+
+=head1 LICENSE
+
+  GPLv2
+
+=head1 Magic markers
+
+  #%# family=contrib
+  #%# capabilities=autoconf
+
+=cut
+
+# Include plugin.sh
+. "${MUNIN_LIBDIR:-}/plugins/plugin.sh"
+is_multigraph "$@"
+
+# Shell options
+set -o nounset
+
+# Global variables
+plugin_name=zpool_capacity
+functions='capacity allocated dedup'
+zpool_cmd=/sbin/zpool
+zfs_cmd=/sbin/zfs
+
+# Environment variables
+: "${warning:=80}"
+: "${critical:=90}"
+
+# Note: The performance of ZFS may significantly degrade when zpool capacity > 90%
+# See also: https://docs.oracle.com/cd/E53394_01/html/E54801/zfspools-4.html
+
+# Functions
+
+preconfig() {
+    local func=$1
```

Comment:
Please add quotes for this kind of `local` assignments. They can cause whitespace issues in some shells (e.g. dash):
```shell
$ foo() { local bar=$1; echo "'$bar' != '$1'"; }; foo "baz bat"
'baz' != 'baz bat'
```